src/orion/core/io/cmdline_parser.py

```diff
@@ -20,29 +20,102 @@
 
 
 class CmdlineParser(object):
-    """CmdlineParser is aimed at providing a simple tool to interpret commandline arguments
+    """Simple class for commandline arguments parsing.
+
+    CmdlineParser aims at providing a simple class to interpret commandline arguments
     for the purposes of Orion. It can transform a list of string representing arguments to their
     corresponding values. It can also recreate that string from the values by maintaing a template
     of the way the arguments were passed.
+
+    Parameters
+    ----------
+    parsing_tokens : dict, optional
+        Dictionary containing the tokens to use when parsing `-` and `_`.
+
+    Attributes
+    ----------
+    arguments : OrderedDict
+        Commandline arguments' name and value(s).
+    template : list
+        List of template-ready strings (in the form 'something_{0}') for formatting.
+
     """
 
-    def __init__(self):
-        """Initialize the OrderedDict of arguments and the template list"""
+    def __init__(self, parsing_tokens=None):
+        """See `CmdlineParser` description"""
         self.arguments = OrderedDict()
         self._already_parsed = False
         self.template = []
 
+        if parsing_tokens is None:
+            parsing_tokens = {'-': '??', '_': '!!'}
```

Comment:
It would be better to use update so that user can pass incomplete dicts. 

```
self.parsing_tokens = {'-': '??', '_': '!!'}
if parsing_tokens it not None:
    self.parsing_tokens.update(parsing_tokens)
```